controllers/volume/volume_controller.go

```diff
@@ -532,6 +533,14 @@ func (r *Reconciler) processAttachments(
 			continue
 		}
 
+		// Skip process attachments if the image hardware version is not at least minimal version.
+		if !conditions.IsTrue(ctx.VM, vmopv1alpha1.PVCHardwareVersionCondition) {
+			ctx.Logger.Info("VirtualMachineImage has an unsupported hardware version for PersistentVolumes. " +
+				"Deferring volume attachment")
+			// Do not return error here since this error is unrecoverable until the image hardware version is updated.
+			return nil
+		}
```

Comment:
Can we also emit an event to the `ctx.VM` to make it more transparent for users?